tests/std/tests/P0645R10_text_formatting_formatting/test.cpp

```diff
@@ -8,252 +8,359 @@
 #include <string>
 #include <string_view>
 
+// copied from the string_view tests
+template <typename CharT>
+struct choose_literal; // not defined
+
+template <>
+struct choose_literal<char> {
+    static constexpr const char* choose(const char* s, const wchar_t*) {
+        return s;
+    }
+};
+
+template <>
+struct choose_literal<wchar_t> {
+    static constexpr const wchar_t* choose(const char*, const wchar_t* s) {
+        return s;
+    }
+};
+
+#define TYPED_LITERAL(CharT, Literal) (choose_literal<CharT>::choose(Literal, L##Literal))
+
+
 using namespace std;

```

Comment:
Nitpick: Usually we put the using-directive immediately after the includes, with no code in the middle.